src/utils.ts

```diff
@@ -31,9 +31,9 @@ export const asNumber = <T>(value: T) => {
 	return integer === float ? integer : float;
 };
 
-export const asDate = <T = any>(value?: T, format?: string): Moment.Moment => {
+export const asDate = <T = any>(value?: T, format?: string): moment.Moment => {
 	// use strict mode of moments
-	return moment(value, format, true);
+	return moment.utc(value, format, true);
```

Comment:
Input types `date`, `month`, `week`, `time` and `datetime-local` do not automatically provide UTC compliant values, thus unspecified input should be treated as local (hence `moment(value, format, true)`